devito/finite_differences/tools.py

```diff
@@ -157,6 +158,11 @@ def symbolic_weights(function, deriv_order, indices, dim):
             for j in range(0, len(indices))]
 
 
+@cacheit
```

Comment:
? why cache it and why not our memoized?

Reply:
good spot, there was an obvious issue with our memoization that I know have forgotten, I should have added a comment...